### PR TITLE
chore: add test for reaction role

### PR DESCRIPTION
**Pull request by haongo138**
**What does this PR do?**

-   [x] Add new format testing for reaction roles

### CODE REVIEW

**Review by trkhoi — APPROVED** [review]

**Review by trkhoi — APPROVED** [review]

**Review by hoangharry — APPROVED** [review]

File: src/commands/reaction-role/list/processor.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        .catch(() => null)
+
+      if (!reactMessage) {
+        throw new InternalError({
+          message: msg,
+          description: "Message not found",
+        })
+      }
+
+      if (cfg.roles && cfg.roles.length > 0) {
```
Comment (anhnh12) [inline src/commands/reaction-role/list/processor.ts:183]:
```suggestion
      if (cfg.roles?.length) {
```

File: src/commands/reaction-role/list/processor.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import { getEmoji, getEmojiURL, emojis, paginate } from "utils/common"
+import { PREFIX } from "utils/constants"
+import {
+  ResponseRole,
+  ResponseRoleReactionByMessage,
+} from "./../../../types/api"
+import { ReactionRoleListConfigItem } from "types/common"
+
+export const handleRoleList = async (msg: Message | CommandInteraction) => {
+  if (!msg.guildId || !msg.guild) {
```
Comment (anhnh12) [inline src/commands/reaction-role/list/processor.ts:24]: no test case for this

File: src/commands/reaction-role/set/processor.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  const requestData: RoleReactionEvent = {
+    guild_id: guildId,
+    message_id: reactMessage.id,
+    reaction,
+    role_id: roleId,
+    channel_id: channelId,
+  }
+
+  const res = await config.updateReactionConfig(requestData)
+  if (!res.ok) {
```
Comment (anhnh12) [inline src/commands/reaction-role/set/processor.ts:30]: this could be any error, at least check error status

**Review by anhnh12 — CHANGES_REQUESTED** [review]

File: src/types/common.ts [hunk 4]
```diff
@@ -182,6 +182,13 @@ export type RoleReactionConfigResponse = {
   success: boolean
 }
 
+export type ReactionRoleListConfigItem = {
+  role?: string
+  emoji?: string
+  url?: string
+  title?: string
+}
+
```
Comment (anhnh12) [inline src/types/common.ts:191]: all API repsonse models should be generated inside `api.ts`
-> remove this
-> remove all reactions role models inside `common.ts`
-> update models in reaction role commands

**Review by anhsetel — APPROVED** [review]

**Review by anhnh12 — APPROVED** [review]

**Pull request merged**